lib/package.gi

```diff
@@ -241,6 +241,30 @@ BindGlobal( "AddPackageInfos", function( files, pkgdir, ignore )
           elif IsRecord( record.PackageDoc ) then
             record.PackageDoc:= [ record.PackageDoc ];
           fi;
+
+          # Normalize the format of 'Date', i.e. if it is the format yyyy-mm-dd
+          # then we change it to dd/mm/yyyy. When other tools have adapted to
+          # the yyyy-mm-dd format we can normalize to that format and at some
+          # point in the future get rid of this code.
+          if record.Date{ [5,8]} = "--" then
+            date := List( SplitString( record.Date, "-" ), Int);
+            date := Permuted(date, (1,3)); # date = [dd,mm,yyyy]
+            # generate the day and month strings
+            # if the day has only one digit we have to add a 0
+            if date[1] < 10 then
+              dd := Concatenation("0", String(date[1]));
+            else
+              dd := String(date[1]);
+            fi;
+            # if the month has only one digit we have to add a 0
```

Comment:
If month has only one digit, this will never be caught by `record.Date{ [5,8]} = "--" ` condition, so a date line `2020-4-21` will not be changed to `21/04/2020` at all.

Reply:
Couldn't be just do `'-' in record.Date`?

Reply:
A date like `2020-4-21` is not a valid ISO date and thus should be rejected, as this PR does.

We still need this code here because above, we convert the parts of the day into integers. A simple implementation of course would be to not do that, and just keep the strings as they are. The calls to `Int` of course validate that we are looking at numbers; but a better check for that might be to change `if record.Date{ [5,8]} = "--" then` by adding something like `... and ForAll(record.Date{1,2,3,4,6,7,9,10}, IsDigit)` . But this starts to feel a bit like bikeshedding ;-)

Reply:
My worry was if someone tried to use the new date format, and messed it up, GAP would get confused by something like 2020-4-21 and try to interpret it as an old format date.

However, on inspection ValidatePackageInfo is checking properly, and we already let any old rubbish in the Date field.